### PR TITLE
Fix/regenerate maneuvers on map update

### PR DESCRIPTION
**Pull request by msmcconnell**
# PR Details
## Description

This PR does two things. 
First, it fixes a bug in carma_wm where the user defined map update callback would only trigger when an entirely new map was received. Now the callback will be triggered either when a new map is received or an update is applied successfully. 

Second, it uses this callback to trigger a maneuver regeneration in the route_following_plugin if a route is also available. The previous temporary logic to handle this case was also removed per discussion with @adev4a 

## Related Issue
https://github.com/usdot-fhwa-stol/carma-platform/issues/1351

## Motivation and Context
Working basic travel use case

## How Has This Been Tested?
build and unit testing

## Types of changes

- [x] Defect fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
[CARMA Contributing Guide](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) 
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by adev4a — COMMENTED** [review]
UpdateEndingSpeed and findLaneletIndexFromPath functions are unused after these changes and should be removed. findLaneletIndexFromPath should also be removed from unit tests

**Comment by msmcconnell** [comment]
> UpdateEndingSpeed and findLaneletIndexFromPath functions are unused after these changes and should be removed. findLaneletIndexFromPath should also be removed from unit tests

Updated and unit tests are passing

File: route_following_plugin/src/route_following_plugin.cpp [hunk 1]
```diff
@@ -59,6 +59,12 @@ namespace route_following_plugin
             this->latest_maneuver_plan_ = routeCb(wm_->getRoute()->shortestPath());
         });
 
+        wml_->setMapCallback([this]() {
+            if (wm_->getRoute()) { // If this map update occured after a route was provided we need to regenerate maneuvers
+                this->latest_maneuver_plan_ = routeCb(wm_->getRoute()->shortestPath());
```
Comment (adev4a) [inline route_following_plugin/src/route_following_plugin.cpp:64]: Do we want a log statement here as well? To indicate what callback is causing the maneuver plan to be replanned

Comment (msmcconnell) [reply route_following_plugin/src/route_following_plugin.cpp:65]: Done

**Pull request merged**